### ISSUE TITLE
Test on other classification datasets

### ISSUE THREAD

**Issue by anubhav2901** [media-comp/2022-Flooding#5]
Try to implement Flooding method for other classification problems and compare the performance of different classifiers.

**Comment by joshua-shuhan** [comment]
Hi! I am trying to implement the flooding method for CIFAR10. 
Is there any strict requirement for classification performance?

**Comment by anubhav2901** [comment]
Hello Joshua, 
There isn't any strict requirement for the classification performance. I think it would be great if we can draw a comparison between the performance of classifiers trained with and without flooding method. 

For your reference,
    In the paper, authors have used ResNet44 for CIFAR-10 and have reported best accuracy of 89.48% using Data Augmentation and Learning rate decay. 

Data Augmentation: Random-crop, Horizontal-flip
Learning rate decay: LR*0.1 after 250th and 400th epoch. 
Training-Validation split: 85:15
Flooding Constant: {0.00, 0.01, .., 0.10}

You may try different classifiers too and let's compare the advantage of using Flooding method.

**Comment by joshua-shuhan** [comment]
Hi! Sorry for my late reply. A little bit busy and spent some time finding a GPU platform.
I use the ResNet18 trained on the CIFAR-10 dataset. For 100 epochs of training, the final accuracy is 90.6%.
I applied the following DL techniques in addition to ResNet18 architecture:

-Data augmentation: Random-crop, Horizontal-flip
-Dropout in the last linear layer of ResNet18 (dropout_rate=0.5)
-Xavier initialization for all layers in ResNet18
-Learning rate is 0.01 with weight_decay=1e-5, momentum=0.9 and `MultiStepLR` scheduler.

Here is the figure for loss comparison. The flooding constant is set as 0.1. 
![image](https://user-images.githubusercontent.com/54022159/170939904-5f26a581-c823-48bf-93b6-264007576aeb.png)
Notice that the test loss w/o flooding didn't go up. This can be the result of dropout and other techniques preventing overfitting. However, the test loss w/ flooding is lower than the test loss w/o flooding, which confirms the effect of the flooding constant.

Here is the figure for accuracy comparison.
![image](https://user-images.githubusercontent.com/54022159/170940651-8d18f64e-2894-4f41-98d4-7915d63a10eb.png)

**Comment by anubhav2901** [comment]
Hi Joshua,

Thanks for the effort. Results are really good and evident that the Flooding method gives an advantage. So submit a pull request for the code, I will merge it with the project.

**Comment by joshua-shuhan** [comment]
Thank you! PR created.
